external/llvm-project/mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td

```diff
@@ -106,18 +106,11 @@ def ROCDL_BarrierOp : ROCDL_Op<"barrier"> {
 def ROCDL_LDSBarrierOp : ROCDL_Op<"lds_barrier"> {
   string llvmBuilder = [{
     llvm::LLVMContext &llvmContext = builder.getContext();
-    llvm::FunctionType *function_type = llvm::FunctionType::get(
-        llvm::Type::getVoidTy(llvmContext), // return type.
-        false);                             // no variadic arguments.
-    llvm::Value *inlineAsm = llvm::InlineAsm::get(
-        function_type,                       // FunctionType *Ty,
-        "s_waitcnt lgkmcnt(0) \n s_barrier", // const std::string &AsmString,
-        "",                                  // const std::string &Constraints,
-        true,                                // bool hasSideEffects,
-        false,                               // bool isAlignStack,
-        llvm::InlineAsm::AD_ATT              // AsmDialect asmDialect
-        );
-    builder.CreateCall(function_type, inlineAsm);
+    //    builder.CreateFence(llvm::AtomicOrdering::Release,
+    //                       llvmContext.getOrInsertSyncScopeID("workgroup"));
```

Comment:
@jungpark-mlir now the PR has got a performance run and looks fine, we should consider land it.

do you think we should use LLVM fence with workgroup synscope, like what you commented here, or directly create intrinsic calls, like what's being implemented below?

Reply:
That's good point, I felt compiler team prefers to use fence. I'll change to it, generate instruction should be identical and will not change the performance.